docs/how-to-use-minio-server-side-encryption-with-aws-cli.md

```diff
@@ -1,88 +1,94 @@
-    ```
-    aws s3api put-object \
-    --no-verify-ssl \
-    --endpoint-url https://localhost:9000 \
-    --bucket my-bucket --key my-secret-diary \
-    --sse-customer-algorithm AES256 \
-    --sse-customer-key MzJieXRlc2xvbmdzZWNyZXRrZXltdXN0cHJvdmlkZWQ= \
-    --sse-customer-key-md5 7PpPLAK26ONlVUGOWlusfg== \
-    --body ~/my-diary.txt
-    ```
-    You should use your own encryption key.
-
-### 2.2 Show object information
-  You **must** specify the correct SSE-C key of an encrypted object to show its metadata:
-  ```
-  aws s3api head-object \
+# How to use Minio's Server-side Encryption with the AWS CLI
+
+Minio supports S3 server-side-encryption with customer provided keys (SSE-C). This following sections describe this use of this encryption with the AWS Command Line Interface (`awscli`):
+* [Prerequisites](#prerequisites)
+* [Use SSE-C with aws-cli](#use-sse-c-with-aws-cli)
+* [Security Notice](#security-notice)
+
+# <a name="prerequisites"></a>Prerequisites
+
+A client must specify three HTTP headers for SSE-C requests:
+* `X-Amz-Server-Side-Encryption-Customer-Algorithm`: The algorithm identifier: must be set to `AES256`.
+* `X-Amz-Server-Side-Encryption-Customer-Key`: The secret encryption key: must be a 256-bit Base64-encoded string.
+* `X-Amz-Server-Side-Encryption-Customer-Key-MD5`: The encryption key MD5 checksum: must be set to the MD5-sum of the encryption key. Note: The MD5 checksum is the MD5 sum of the raw binary key, not of the base64-encoded key.
+
+Install the Minio Server with TLS as described [here](https://docs.minio.io/docs/how-to-secure-access-to-minio-server-with-tls).
+
+**Note**: Tools like `aws-cli` or `mc` will display an error if a self-signed TLS certificate is used when trying to upload objects to the server. See [Let's Encrypt](https://letsencrypt.org/) to get a CA-signed TLS certificate. Self-signed certificates should only be used for development, testing or internal usage.
+
+# <a name="use-sse-c-with-aws-cli"></a>Use SSE-C with aws-cli
+
+This section describes how to use server-side encryption with customer-provided encryption (SSE-C) keys via the aws-cli.
+
+## Install the aws-cli 
+You can install the AWS Command Line Interface using the procedure described [here](https://docs.minio.io/docs/aws-cli-with-minio).
+
+## Create a bucket named `my-bucket`
+
+```sh
+aws --no-verify-ssl --endpoint-url https://localhost:9000 s3api create-bucket --bucket my-bucket
+```
+
+## Upload an Object using SSE-C
+
+The following example shows how to upload an object named `my-secret-diary` where the content is the file: `~/my-diary.txt`:
+
+```sh
+aws s3api put-object \
+  --no-verify-ssl \
+  --endpoint-url https://localhost:9000 \
+  --bucket my-bucket --key my-secret-diary \
+  --sse-customer-algorithm AES256 \
+  --sse-customer-key MzJieXRlc2xvbmdzZWNyZXRrZXltdXN0cHJvdmlkZWQ= \
+  --sse-customer-key-md5 7PpPLAK26ONlVUGOWlusfg== \
+  --body ~/my-diary.txt You should use your own encryption key.
```

Comment:
"...You should use your own encryption key."  This should be outside the code block

Reply:
@eco-minio   I have revised as follows: 

The following example shows how to upload an object named `my-secret-diary` where the content is the file `~/my-diary.txt`. Note that you should use your own encryption key.

```sh
aws s3api put-object \
  --no-verify-ssl \
  --endpoint-url https://localhost:9000 \
  --bucket my-bucket --key my-secret-diary \
  --sse-customer-algorithm AES256 \
  --sse-customer-key MzJieXRlc2xvbmdzZWNyZXRrZXltdXN0cHJvdmlkZWQ= \
  --sse-customer-key-md5 7PpPLAK26ONlVUGOWlusfg== \
  --body ~/my-diary.txt 
```